### PR TITLE
[WIP] Selfie extension

### PR DESCRIPTION
**Pull request by sedge**
This is pending user testing, reducing code duplication and a cute cat image.

### CODE REVIEW

File: src/extensions/default/ImageUrlCodeHints/camera.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        var context = canvas.getContext("2d");
+        var data;
+
+        function savePhoto() {
+            if(!data) {
+                return deferred.reject();
+            }
+
+            var bin = /^data:image\/png;base64,(.+)/.exec(data)[1];
+            save(deferred, filePath, base64ToBuffer(bin));
```
Comment (sedge) [inline src/extensions/default/ImageUrlCodeHints/camera.js]: There's some weirdness in this file, like here where `deferred` and `filePath` are actually global and don't need to be passed to `save()`.

File: src/utils/ExtensionLoader.js [hunk 2]
```diff
@@ -1,24 +1,24 @@
 /*
  * Copyright (c) 2012 Adobe Systems Incorporated. All rights reserved.
- *
```
Comment (sedge) [inline src/utils/ExtensionLoader.js]: This needs to be backed out. I bet that's me.

File: src/extensions/default/ImageUrlCodeHints/style.less [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    width:220px;
+    height:180px;
+}
+ 
+.polaroid p {
+  position: absolute;
+  text-align: center;
+  width: 100%;
+  bottom: 0px;
+  font: 400 18px/1 'Kaushan Script', cursive;
```
Comment (humphd) [inline src/extensions/default/ImageUrlCodeHints/style.less]: Hm, not sure I love this.

**Comment by sedge** [comment]
To be fixed:

* [ ] Disable "select selfie" button until a picture has been taken
* [ ] Disable "take picture" button until video stream has activated
* [ ] Correct preview aspect ratio

**Pull request closed without merging**